plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/DefaultLocationProvider.kt

```diff
@@ -26,10 +26,10 @@ class DefaultLocationProvider @VisibleForTesting(otherwise = PRIVATE) internal c
   context: Context,
   private val locationCompassEngine: LocationCompassEngine
 ) : LocationProvider {
-  constructor(context: Context) : this(context, LocationCompassEngine(context))
+  constructor(context: Context) : this(context, LocationCompassEngine(context.applicationContext))
 
-  private val contextWeekRef: WeakReference<Context> = WeakReference(context)
-  private val locationEngine = LocationEngineProvider.getBestLocationEngine(context)
+  private val applicationContext = context.applicationContext
```

Comment:
nit: would it be safer to keep a weak reference instead of a strong reference?

Reply:
Why do we need weak reference though? ApplicationContext is safe to keep forever since it lives until the application is finished. IMO WeakReference here brings more confusion making us to think it's safe to keep any context including activity.